Update content-style-guide.md

mirnawong1 opened the pull request:
updates to the style guide

## Description & motivation
<!---
Describe your changes, and why you're making them. Is this linked to an open
issue, a pull request on dbt core, etc?

To learn more about the writing conventions used in the dbt Labs docs, see the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md).
-->

## To-do before merge
<!---
(Optional -- remove this section if not needed)
Include any notes about things that need to happen before this PR is merged, e.g.:
- [ ] Change the base branch
- [ ] Ensure PR #56 is merged
-->

## Prerelease docs
If this change is related to functionality in a prerelease version of dbt (delete if not applicable):
- [ ] I've added versioning components, as described in ["Versioning Docs"](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/versioningdocs.md)
- [ ] I've added a note to the prerelease version's [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)

## Checklist
If you added new pages (delete if not applicable):
- [ ] The page has been added to `website/sidebars.js`
- [ ] The new page has a unique filename

If you removed existing pages (delete if not applicable):
- [ ] The page has been removed from `website/sidebars.js`
- [ ] An entry has been added to `_redirects`
- [ ] [Run link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

mirnawong1 committed:
Update content-style-guide.md

moved most of the readme edits to the style guide per [docs sync meeting here](https://www.notion.so/dbtlabs/Docs-team-meeting-6fcdc2fffb374ba7b7e095076a1d852a). there are some outstanding questions i have and below are the lines where i flagged more input from the team:

- line 60 - added a callout format table since there's nothing detailed in github or other public documentation. i also added the <br /> because i believe you need the spacing right for callouts. 
- line 205 - do we have any guidance or preferences as a team, on file naming conventions?
- line 406 - is png and jpg what we accept?

mshaver631 commented on contributing/content-style-guide.md:
```diff
@@ -402,7 +442,12 @@ Once you have determined that a screenshot will add value to the document where
 * Avoid using screenshots to demonstrate inputs and outputs. All code snippets and sample results should be in the documents as text fields.
+* Add images are under the `static` -> `img` folder.
 * Use concise filenames that are relevant to the content contained within. Enumerate them if they are part of a sequence.
+* Use PNG or JPEG format, which renders a better quality and lossless compression.
```
Should we remove PNG based on the feedback around dark mode support?

mirnawong1 replied:
i think so too! i posed that question up top in the description and IMO it makes sense as the win-win option but def wanted to double check!

mshaver631 commented on contributing/content-style-guide.md:
```diff
@@ -402,7 +442,12 @@ Once you have determined that a screenshot will add value to the document where
+* Add images are under the `static` -> `img` folder.
 * Use concise filenames that are relevant to the content contained within. Enumerate them if they are part of a sequence.
+* Use PNG or JPEG format, which renders a better quality and lossless compression.
+* Add *title=""* for all images to write a consice title of the image. For accessibility, it's important to use succinct text that is clear and complete. 
```
```suggestion
* Add *title=""* for all images to write a concise title of the image. For accessibility, it's important to use succinct text that is clear and complete. 
```

mshaver631 commented on contributing/content-style-guide.md:
```diff
@@ -18,11 +18,51 @@ This guide includes standards we want to emphasize, likely because we've made de
+
+## Filenaming
+
+If you are adding a new file in the [docs.getdbt.com](https://github.com/dbt-labs/docs.getdbt.com) repository, it must be named following our naming conventions. MORE TO ADD
```
Should we define our naming conventions before we add this line (remove the "MORE TO ADD")?

mirnawong1 replied:
yes def! we can try to decide on this async but that's another one of my questions in the description up top!  maybe i can start a thread in slack to get it started?

mirnawong1 replied:
Ideas for content once polling is finished:

When you create or rename a file or directory and it has more than one word in its name, use dashes (-) instead of spaces or underscores. 

**Option 1:**
The file name should always start with a verb and the feature type, like ide, scheduler, or python. Depending on the content type, it typically also includes a secondary descriptor. Verbs are used when you are creating a procedural document. 

Due to the fact that we author content using Markdown, you must add the `.md` file extension to the file name. For example, `ide-features.md`

If you are adding a new file to an existing feature category, follow the current naming convention.

| File type | Example |
| :-------: | :-------: |
| Conceptual file | `set-up-dbt-cloud.md` |
| Procedural file | `set-up-cloud-environment.md` |
| Reference file | `exclude-flag.md` |

**Option 2:**
The file name should always start with the feature type, like ide, scheduler, or python. Depending on the content type, it typically also includes a secondary descriptor or a verb. Verbs are used when you are creating a procedural document. 

Due to the fact that we author content using Markdown, you must add the `.md` file extension to the file name. For example, `ide-features.md`

If you are adding a new file to an existing feature category, follow the current naming convention.

| File type | Example |
| :-------: | :-------: |
| Conceptual file | `dbt-cloud-set-up.md` |
| Procedural file | `cloud-environment-set-up.md` |
| Reference file | `exclude-flag.md` |

mirnawong1 committed:
Update contributing/content-style-guide.md

omg great catch 🙈

Co-authored-by: mshaver631 <118219367+mshaver631@users.noreply.github.com>

mshaver631 commented on contributing/content-style-guide.md:
```diff
@@ -18,11 +18,51 @@ This guide includes standards we want to emphasize, likely because we've made de
+
+If you are adding a new file in the [docs.getdbt.com](https://github.com/dbt-labs/docs.getdbt.com) repository, it must be named following our naming conventions. MORE TO ADD
+
+## Using Markdown
```
This whole section is awesome and very useful.  If/when we rearrange the order of these sections, I think this should be at the top.

mirnawong1 committed:
Merge branch 'current' into mwong-styleguide-add

mirnawong1 committed:
Update content-style-guide.md

mirnawong1 committed:
Update content-style-guide.md

mirnawong1 committed:
Update content-style-guide.md

mirnawong1 committed:
Merge branch 'current' into mwong-styleguide-add

mirnawong1 committed:
Update content-style-guide.md

mirnawong1 committed:
Merge branch 'current' into mwong-styleguide-add

runleonarun commented on contributing/content-style-guide.md:
```diff
@@ -18,11 +18,60 @@ This guide includes standards we want to emphasize, likely because we've made de
 Callouts highlight important or high-value information that readers need to know. We want callouts to stand out, so we should keep their content to a minimum, avoiding general information, permissions, or prerequisites. Too much information can make it difficult to absorb. Imagine driving down one block with five stop signs!?!
 
-Use callouts sparingly for high-value information &mdash; avoid including general information, permissions, or prerequisites in callouts.
+Use callouts sparingly for high-value information &mdash; avoid including general information, permissions, or prerequisites in callouts. Below are the some of the typical callout formats we use:
```
This looks great! We might also want to say that if you add text after the first code, like this: `:::note This is a note title` it shows up as a title for the note.

runleonarun commented on contributing/content-style-guide.md:
```diff
@@ -278,6 +327,11 @@ A table following an H3 heading:
 > | `-file` | Print output to file instead of stdout. | Name of the file. |
 
 ## Word choice & terminology
+Use active voice instead of passive. Active voice is clearer and more direct, making it easier to translate. 
```
GREAT addition! 🙇🏻‍♀️

runleonarun commented on contributing/content-style-guide.md:
```diff
@@ -438,7 +492,12 @@ Once you have determined that a screenshot will add value to the document where
+* Add images are under the `static` -> `img` folder.
 * Use concise filenames that are relevant to the content contained within. Enumerate them if they are part of a sequence.
+* Use JPEG format, which renders a better quality and lossless compression. JPEG format has a white background and is accessible in light and dark mode. 
+* Add *title=""* for all images to write a concise title of the image. For accessibility, it's important to use succinct text that is clear and complete. 
```
Should we move [this content](https://www.notion.so/dbtlabs/Lightbox-Component-19d4496c073542868dee7bf33ff59b30) to the `contributing/` folder so we can link to it?

mirnawong1 replied:
good callout!

runleonarun commented on contributing/content-style-guide.md:
```diff
@@ -18,11 +18,60 @@ This guide includes standards we want to emphasize, likely because we've made de
+
+| ✅ User | ❌ Avoid|
+|:-------:|:-------:|
+|`/docs/deploy/architecture` | `/docs/deploy/about-architecture`|
```
I would be OK with `about-architecture.md` maybe a better thing to avoid is something like `how-we-think-about-architecture-at-dbt.md` 😆

mirnawong1 replied:
thanks @runleonarun , oh gotcha!  i used one of @nghi-ly 's suggestions on this one but maybe it makes sense to include that example you shared too 🙈

mirnawong1 committed:
Merge branch 'current' into mwong-styleguide-add

mirnawong1 commented on contributing/content-style-guide.md:
```diff
@@ -18,11 +18,60 @@ This guide includes standards we want to emphasize, likely because we've made de
+
+| ✅ User | ❌ Avoid|
+|:-------:|:-------:|
+|`/docs/deploy/architecture` | `/docs/deploy/about-architecture`|
```
```suggestion
|`/docs/deploy/architecture` | `/docs/deploy/how-we-think-about-architecture`|
```

mirnawong1 committed:
Update contributing/content-style-guide.md

mirnawong1 committed:
Update content-style-guide.md

mirnawong1 committed:
Create lightbox.md

mirnawong1 committed:
Update lightbox.md

mirnawong1 committed:
Update content-style-guide.md

mirnawong1 commented:
thanks @runleonarun ! i've made those changes and it should be ready for review again 🙏

nghi-ly committed:
Merge branch 'current' into mwong-styleguide-add

nghi-ly commented on contributing/content-style-guide.md:
```diff
@@ -18,11 +21,62 @@ This guide includes standards we want to emphasize, likely because we've made de
 
+## Folder Structure and TOC
+
+The folder structure for the [docs.getdbt.com](https://github.com/dbt-labs/docs.getdbt.com) repository is organized into several high-level categories under the main `website` folder: blog, cypress, docs, functions, plugins, snippets, src, static.
```
```suggestion
The folder structure for the [docs.getdbt.com](https://github.com/dbt-labs/docs.getdbt.com) repository is organized into several high-level categories under the main `website` folder such as `blog`, `cypress`, `docs`, `functions`, `plugins`, `snippets`, `src`, `static`, and so on.
```

runleonarun commented on contributing/content-style-guide.md:
````diff
@@ -18,11 +21,62 @@ This guide includes standards we want to emphasize, likely because we've made de
+| Tip callouts are used for tips |```:::tip``` <br /> <br /> ```text``` <br /> <br /> ```:::``` |
+| Caution callouts are used for warnings/considerations |```:::caution``` <br /> <br /> ```text``` <br /> <br /> ```:::``` |
+
+Note: If you add text after the first code, like this `:::note This is a note title`, it will show up as a title for the note.
````
Just moving this to the top so it doesn't get lost.
```suggestion
```

runleonarun committed:
Update contributing/content-style-guide.md

nghi-ly commented on contributing/content-style-guide.md:
```diff
@@ -18,11 +21,62 @@ This guide includes standards we want to emphasize, likely because we've made de
+
+The folder structure for the [docs.getdbt.com](https://github.com/dbt-labs/docs.getdbt.com) repository is organized into several high-level categories under the main `website` folder: blog, cypress, docs, functions, plugins, snippets, src, static.
+
+The table of contents (TOC) is managed in the `sidebar.js` file. You only need to edit the `sidebar.js` file when you are adding a new page or deleting an existing page. The `sidebar.js` file is the one that causes most of the merge conflicts because many technical writers are working on content daily. You will need to accept the changes from other contributors if you are committing a PR.
```
```suggestion
The left sidebar (leftbar) is managed in the `sidebar.js` file. You only need to edit the `sidebar.js` file when you are adding a new page or deleting an existing page. Editing the `sidebar.js` file can cause merge conflicts as it's updated often because content is being worked on daily. You will need to accept the changes from other contributors if you are committing a PR.
```

nghi-ly commented on contributing/content-style-guide.md:
```diff
@@ -18,11 +21,62 @@ This guide includes standards we want to emphasize, likely because we've made de
+
+The table of contents (TOC) is managed in the `sidebar.js` file. You only need to edit the `sidebar.js` file when you are adding a new page or deleting an existing page. The `sidebar.js` file is the one that causes most of the merge conflicts because many technical writers are working on content daily. You will need to accept the changes from other contributors if you are committing a PR.
+
+Don't worry if you're not sure where in the TOC a new topic belongs. Do your best and when you submit your PR, the dbt Labs Documentation team will edit it and help to find the right placement.
```
```suggestion
Don't worry if you're not sure where in the leftbar a new topic belongs. Do your best and when you submit your PR, the dbt Labs Documentation team will edit it and help to find the right placement.
```

nghi-ly commented on contributing/content-style-guide.md:
```diff
@@ -18,11 +21,62 @@ This guide includes standards we want to emphasize, likely because we've made de
+
+Don't worry if you're not sure where in the TOC a new topic belongs. Do your best and when you submit your PR, the dbt Labs Documentation team will edit it and help to find the right placement.
+
+The right-hand TOC is created automatically when you add headings to a page.
```
```suggestion
The right sidebar (rightbar), also known as the page TOC, is created automatically for all the H2 and H3 headings on a page. For details, see [Headings and Table of contents](https://docusaurus.io/docs/markdown-features/toc) in Docusaurus. 
```

nghi-ly commented on contributing/content-style-guide.md:
```diff
@@ -18,11 +21,61 @@ This guide includes standards we want to emphasize, likely because we've made de
+| Link - section in topic in same folder      | `[Title](/folder/file-name#section-name)`                     |
+| Link - section in topic in different folder | `[Title](/folder/file-name#section-name)`           |
+| Image                                       | `<Lightbox src="/img/docs/<image-name>.jpg" title="Concise description of image"/>`      |
+
```
sugg:

- linking to https://www.markdownguide.org/basic-syntax/ for basic Markdown syntax instead of listing them in the table

reason: 

- for conciseness and easier docs maintenance. not too lengthy of a style guide and changes to basic Markdown are maintained elsewhere 
- to have content be focused on what's different or custom with ours ("our stuff"). include things like: internal links should be in the format of "/docs/<path-to-file>" and Lightbox (which is custom, i think) and so on.

nghi-ly commented on contributing/content-style-guide.md:
```diff
@@ -18,11 +21,61 @@ This guide includes standards we want to emphasize, likely because we've made de
+| Unordered List                              | `-` or `*` (for each item)                            |
+| Code or command in a sentence               | ``code``                                              |
+| Link - external site                        | `[Title](https://www.example.com)`                    |
+| Link - topic in same folder                 | `[Title](/folder/file-name) without file extension`            |
```
i don't believe we want to use this ^ link format. i think we want the "/docs/<path-to-file>" format

mirnawong1 replied:
ah ok gotcha, i kept it generic as '/folder/file-name' as it's different for guides and other folders (unless that still starts with /docs/guides...?)

I've changed it to /docs though but will bring it up at our session

nghi-ly commented on contributing/content-style-guide.md:
```diff
@@ -18,11 +21,61 @@ This guide includes standards we want to emphasize, likely because we've made de
+| Code or command in a sentence               | ``code``                                              |
+| Link - external site                        | `[Title](https://www.example.com)`                    |
+| Link - topic in same folder                 | `[Title](/folder/file-name) without file extension`            |
+| Link - topic in different folder            | `[Title](/folder/file-name) without file extension` |
```
i think the same thing with this ^

nghi-ly commented on contributing/content-style-guide.md:
```diff
@@ -18,11 +21,61 @@ This guide includes standards we want to emphasize, likely because we've made de
+| Link - external site                        | `[Title](https://www.example.com)`                    |
+| Link - topic in same folder                 | `[Title](/folder/file-name) without file extension`            |
+| Link - topic in different folder            | `[Title](/folder/file-name) without file extension` |
+| Link - section in topic in same folder      | `[Title](/folder/file-name#section-name)`                     |
```
same thing with this ^

nghi-ly commented on contributing/content-style-guide.md:
```diff
@@ -18,11 +21,61 @@ This guide includes standards we want to emphasize, likely because we've made de
+| Link - topic in same folder                 | `[Title](/folder/file-name) without file extension`            |
+| Link - topic in different folder            | `[Title](/folder/file-name) without file extension` |
+| Link - section in topic in same folder      | `[Title](/folder/file-name#section-name)`                     |
+| Link - section in topic in different folder | `[Title](/folder/file-name#section-name)`           |
```
same thing with this ^

nghi-ly commented on contributing/content-style-guide.md:
```diff
@@ -18,11 +21,61 @@ This guide includes standards we want to emphasize, likely because we've made de
 Callouts highlight important or high-value information that readers need to know. We want callouts to stand out, so we should keep their content to a minimum, avoiding general information, permissions, or prerequisites. Too much information can make it difficult to absorb. Imagine driving down one block with five stop signs!?!
 
-Use callouts sparingly for high-value information &mdash; avoid including general information, permissions, or prerequisites in callouts.
+Use callouts sparingly for high-value information &mdash; avoid including general information, permissions, or prerequisites in callouts. Below are the some of the typical callout formats we use:
```
```suggestion
Below are the some of the typical callout formats we use:
```

nghi-ly replied:
sugg: second sentence in previous paragraph mentions this too

runleonarun replied:
I like Ly's sugg and we can even make it shorter because I think these are the only formats possible. 
```suggestion
Callout formats include:
```

mirnawong1 replied:
good shout! the paragraph previously existed and i only added the last sentence. however, i love that we can shorten it!

nghi-ly commented on contributing/content-style-guide.md:
````diff
@@ -18,11 +21,61 @@ This guide includes standards we want to emphasize, likely because we've made de
+| Info callouts are used to highlight info |```:::info``` <br /> <br /> ```text``` <br /> <br /> ```:::``` |
+| Tip callouts are used for tips |```:::tip``` <br /> <br /> ```text``` <br /> <br /> ```:::``` |
+| Caution callouts are used for warnings/considerations |```:::caution``` <br /> <br /> ```text``` <br /> <br /> ```:::``` |
+
````
<3

nghi-ly commented on contributing/lightbox.md:
```diff
@@ -0,0 +1,39 @@
+
+The Lightbox component allows clickable images, which are expanded when clicked.
```
```suggestion
You can use the Lightbox component to add an image or screenshot to your page. It supports clickable images that expand when clicked.
```

nghi-ly commented on contributing/lightbox.md:
```diff
@@ -0,0 +1,39 @@
+- **************alignment************** - Determine if image should be left, center, or right aligned
+    - Optional (Defaults to center)
+    - Accepts `left`, `right`, `center`. If any other value is entered, it defaults to `center`
+
```
sugg using a table instead of a list to present this ^ referential content. lookup might be a bit easier with a table.

nghi-ly reviewed (dismissed):
thank you so much for making our Style and usage guide better! it's going to help so many people! happy to answer any questions that you may have about my review comments!

runleonarun commented on contributing/content-style-guide.md:
```diff
@@ -18,11 +21,61 @@ This guide includes standards we want to emphasize, likely because we've made de
+
 ## Callouts
 
 Callouts highlight important or high-value information that readers need to know. We want callouts to stand out, so we should keep their content to a minimum, avoiding general information, permissions, or prerequisites. Too much information can make it difficult to absorb. Imagine driving down one block with five stop signs!?!
```
Moving the note about the title from below to here. 

```suggestion
Callouts highlight important or high-value information that readers need to know. We want callouts to stand out, so we should keep their content to a minimum, avoiding general information, permissions, or prerequisites. Too much information can make it difficult to absorb. Imagine driving down one block with five stop signs!?!

If you add text after the first code, like this `:::note This is a note title`, it shows up as a title for the note.
```

runleonarun reviewed (dismissed):
Just a few tiny suggestions!

mirnawong1 committed:
Merge branch 'current' into mwong-styleguide-add

mirnawong1 committed:
Merge branch 'current' into mwong-styleguide-add

mirnawong1 committed:
Update contributing/content-style-guide.md

Co-authored-by: Ly Nguyen <107218380+nghi-ly@users.noreply.github.com>

mirnawong1 committed:
Update contributing/content-style-guide.md

Co-authored-by: Ly Nguyen <107218380+nghi-ly@users.noreply.github.com>

mirnawong1 committed:
Update contributing/content-style-guide.md

Co-authored-by: Ly Nguyen <107218380+nghi-ly@users.noreply.github.com>

mirnawong1 committed:
Update contributing/content-style-guide.md

thanks for linking to the docusaurus link!

Co-authored-by: Ly Nguyen <107218380+nghi-ly@users.noreply.github.com>

mirnawong1 committed:
Update contributing/content-style-guide.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Update contributing/lightbox.md

Co-authored-by: Ly Nguyen <107218380+nghi-ly@users.noreply.github.com>

mirnawong1 committed:
Update contributing/content-style-guide.md

Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>

mirnawong1 committed:
Update lightbox.md

mirnawong1 committed:
Merge branch 'current' into mwong-styleguide-add

mirnawong1 committed:
Update content-style-guide.md

mirnawong1 commented:
hey @runleonarun and @nghi-ly - made those changes and thanks so much for the amazing feedback! let me know what ya'll think 🙏

nghi-ly committed:
Merge branch 'current' into mwong-styleguide-add

nghi-ly committed:
Merge branch 'current' into mwong-styleguide-add

nghi-ly committed:
Merge branch 'current' into mwong-styleguide-add

mirnawong1 committed:
Merge branch 'current' into mwong-styleguide-add

mirnawong1 committed:
Update content-style-guide.md

adding 'read' per Leona's comment [here](https://github.com/dbt-labs/docs.getdbt.com/pull/2647#pullrequestreview-1236678831)

mirnawong1 committed:
Merge branch 'current' into mwong-styleguide-add

mirnawong1 committed:
Update content-style-guide.md

mirnawong1 committed:
Update content-style-guide.md

matthewshaver reviewed (dismissed):
Looks like this was already pretty thoroughly reviewed so approving for merge.

runleonarun committed:
Merge branch 'current' into mwong-styleguide-add

mirnawong1 committed:
Merge branch 'current' into mwong-styleguide-add

mirnawong1 committed:
Update content-style-guide.md

mirnawong1 commented:
i'm going to merge this as it looks like it was approved a few weeks ago!
